ci-operator/step-registry/telcov10n/functional/llc-cnf/tests/telcov10n-functional-llc-cnf-tests-commands.sh

```diff
@@ -0,0 +1,196 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+set -x
+
+echo "************ telco verfication functional tests commands ************"
+
+# Environment Variables required for running the test
+export KUBECONFIG="${SHARED_DIR}"/kubeconfig
+export ROLE_WORKER_CNF=worker-cnf
+
+# local variables
+TELCO_CI_REPO="https://github.com/openshift-kni/telco-ci.git"
+NTO_REPO="https://github.com/openshift/cluster-node-tuning-operator.git"
+NTO_BRANCH=$(git ls-remote --heads ${NTO_REPO} main | grep -q 'refs/heads/main'  && echo 'main' || echo 'master')
+GINKGO_LABEL="uncore-cache"
+GINKGO_SUITES="test/e2e/performanceprofile/functests/13_llc"
+
+[[ -f "${SHARED_DIR}"/main.env ]] && source "${SHARED_DIR}"/main.env || echo "No main.env file found"
+
+# Fix user ID's in a container
+~/fix_uid.sh
+
+echo "************ Telco CNF Compute QE Setup Command ************"
+date +%s > "${SHARED_DIR}"/start_time
+
+mkdir -p "${SHARED_DIR}"/repos
+git clone "${TELCO_CI_REPO}" "${SHARED_DIR}"/repos/telco-ci
+
+# install ansible modules
+ansible-galaxy collection install -r "${SHARED_DIR}"/repos/telco-ci/ansible-requirements.yaml
+
+#Find Baremetal worker nodes
+function is_bm_node () {
+    node=$1
+
+    CPU_THRESHOLD=79
+    MEM_THRESHOLD=80 # in Gi (80 GB)
+
+    echo "Check if node ${node} is baremetal or virtual"
+
+    # Get CPU and memory capacity of the nodes
+    cpu=$(oc get ${node} -o jsonpath='{.status.capacity.cpu}')
+    memory=$(oc get ${node} -o jsonpath='{.status.capacity.memory}')
+
+    # Handle both Ki and Mi memory units
+    if [[ ${memory} == *Ki ]]; then
+        memory=${memory%Ki}
+        memory=$((memory / 1024 / 1024 ))  # Convert Ki to Gi
+    elif [[ ${memory} == *Mi ]]; then
+        memory=${memory%Mi}
+        memory=$((memory / 1024 ))  # Convert Mi to Gi
+    else
+        echo "Warning: Unknown memory unit for ${memory}"
+        memory=0
+    fi
+
+    if [[ ${cpu} -gt ${CPU_THRESHOLD} && ${memory} -gt ${MEM_THRESHOLD} ]]; then
+        echo "Node ${node} is a baremetal node"
+        return 0
+    else
+       echo "Node ${node} is a virtual node"
+       return 1
+    fi
+}
+
+# Label Baremetal worker nodes as worker-cnf
+worker_nodes=$(oc get nodes --selector='node-role.kubernetes.io/worker' \
+    --selector='!node-role.kubernetes.io/master' -o name)
+if [ -z "${worker_nodes}" ]; then
+    echo "No worker nodes found"
+    exit 0
```

Comment:
This will exit successfully if no worker nodes were found, perhaps you should exit 1?